### PR TITLE
Login jquery validation

### PR DESCRIPTION
**Pull request by pkakelas**

### CODE REVIEW

File: static/script/form_validation.js [hunk 1]
```diff
@@ -44,6 +44,40 @@ $( document ).ready( function() {
         }
     }); 
 
+    $( '#login-form' ).submit( function() {
+        var form = $("#login-form"); // or $("form"), or any selector matching an element containing your input fields
```
Comment (vitsalis) [inline static/script/form_validation.js]: add an `$` on the names of jquery objects.

**Comment by vitsalis** [comment]
You should move form validation to according folders. like validation for `session/create` `static/script/session/create.js`. That's how we do it for every script.

File: static/script/form_validation.js (hunk 1 above)
Comment (dionyziz) [inline static/script/form_validation.js]: Remove comment copied from tutorial.

Comment (dionyziz) [inline static/script/form_validation.js]: nit: space after `(` and before `)`

File: static/script/form_validation.js [hunk 2]
```diff
@@ -44,6 +44,40 @@ $( document ).ready( function() {
...
+        var token = $( "[name='token']", form ).val();
+        if ( username == '' ) {
+            createError( '#login-form', 'Please type a username' );
+            return false;
+        }
+        if ( password == '' ) {
+            createError( '#login-form', 'Please type a password' );
+            return false;
+        }
+        $.ajax({
```
Comment (dionyziz) [inline static/script/form_validation.js]: nit: space after `(`

File: static/script/form_validation.js [hunk 3]
```diff
@@ -44,6 +44,40 @@ $( document ).ready( function() {
...
+                    createError( '#login-form', "Username doesn't exist" );
+                },
+                401: function() { 
+                    createError( '#login-form', 'Password is incorrect' );
+                },
+                200: function() {
+                    window.location.replace( 'dashboard' );
+                }
+            }
+        });
```
Comment (dionyziz) [inline static/script/form_validation.js]: nit: space before `)`

File: static/script/form_validation.js [hunk 4]
```diff
@@ -44,6 +44,40 @@ $( document ).ready( function() {
...
+                401: function() { 
+                    createError( '#login-form', 'Password is incorrect' );
+                },
+                200: function() {
+                    window.location.replace( 'dashboard' );
+                }
+            }
+        }); 
+        return false;
+    });
```
Comment (dionyziz) [inline static/script/form_validation.js]: nit: space before `)`

File: static/script/form_validation.js [hunk 5]
```diff
@@ -44,6 +44,40 @@ $( document ).ready( function() {
...
+        }
+        $.ajax({ 
+            type: "POST",
+            url: "session/create",
+            data: { username: username, password: password, token: token },
+            dataType: "json",
+            async: true,
+            statusCode: { 
+                404: function() {
+                    createError( '#login-form', "Username doesn't exist" );
```
Comment (dionyziz) [inline static/script/form_validation.js]: Suggest also focusing and selecting the username field.

File: static/script/form_validation.js [hunk 6]
```diff
@@ -44,6 +44,40 @@ $( document ).ready( function() {
...
+                },
+                401: function() { 
+                    createError( '#login-form', 'Password is incorrect' );
```
Comment (dionyziz) [inline static/script/form_validation.js]: Suggest focusing and selecting the password field.

**Comment by dionyziz** [comment]
Please merge from master.

**Pull request merged**